### DISCUSSION TITLE
Open project instead of file when single file is given to `zed` cli

### DISCUSSION THREAD

**Discussion by failable** [zed-industries/zed#26199] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

If the single file is part of a project, `zed that_file` will only open the file but not the project. For example,

```
cargo new project
zed project/src/main.rs
```

will only open `main.rs` in the new window. I think if a *single* path is given `zed` in this case, it may be better to also open the whole project instead. 

Otherwise, one can not select any other files from the sidebar or the file selector (`file finder: toggle`) because they are not listed there. That's a bit inconvenient. 

**(Note that I know `zed project` will open the project in this case, but this FR is not about opening a project but "opening the related project when a single file path is given")**

**Comment by JosephTLyons** [comment]
Hey @liebkne, it is possible I'm missing something here, but how would zed know where to open the project at? For instance, if I provided a full, aboslute path to a file:

`zed /Users/josephlyons/Desktop/some_rust_project/main.rs`

What would it consider to be the project?

**Comment by failable** [comment]
@JosephTLyons That might be the missing part. For example, the [`projectile`](https://docs.projectile.mx/projectile/projects.html#supported-project-types) plugin of Emacs has some mechanisms to detect a project like top down/bottom up identifying `Cargo.toml` or a VC repo.

> Broadly speaking, Projectile identifies projects like this:
> - Directories that contain the special .projectile file
> - Directories under version control (e.g. a Git repo)
> - Directories that contain some project description file (e.g. a Gemfile for Ruby projects or pom.xml for Java maven-based projects)

In the worse situation I think the VC repo could be opened in this case.
